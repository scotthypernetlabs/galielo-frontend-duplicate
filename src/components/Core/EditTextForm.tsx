import { Button, TextField } from "@material-ui/core";
import React from "react";

interface EditTextFormProps {
  name: string;
  handleChange: any;
  handleEditText: any;
  handleDiscardText: any;
}

const EditTextForm: React.SFC<EditTextFormProps> = (
  props: EditTextFormProps
) => {
  const { name, handleChange, handleEditText, handleDiscardText } = props;
  return (
    <div>
      <TextField
        value={name}
        variant="outlined"
        size="small"
        onChange={handleChange}
      />
      <div>
        <Button variant="contained" onClick={handleEditText}>
          Save
        </Button>
        <Button variant="contained" onClick={handleDiscardText}>
          Discard
        </Button>
      </div>
    </div>
  );
};

export default EditTextForm;
