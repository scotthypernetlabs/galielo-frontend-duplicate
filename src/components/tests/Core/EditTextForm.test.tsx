import { Button, TextField } from "@material-ui/core";
import { mount, shallow } from "enzyme";
import EditTextForm from "../../Core/EditTextForm";
import React from "react";

describe("EditTextForm Component", () => {
  const componentProps = {
    name: "Testing",
    handleChange: jest.fn((e: any) => {
      componentProps.name = e.target.value;
    }),
    handleEditText: jest.fn(),
    handleDiscardText: jest.fn()
  };

  const Wrapper = (props: any) => {
    return <EditTextForm {...componentProps} />;
  };

  const editTextForm = mount(<Wrapper />);

  it("text field should call handleChange", () => {
    const newText = "RandomString";
    expect(editTextForm.childAt(0).props().name).toEqual(componentProps.name);
    expect(editTextForm.find(TextField)).toHaveLength(1);
    editTextForm
      .find("input")
      .simulate("change", { target: { value: newText } });
    expect(componentProps.handleChange).toHaveBeenCalled();
    expect(componentProps.name).toEqual(newText);
  });

  it("button 1 should call handleEditText", () => {
    expect(editTextForm.find(Button)).toHaveLength(2);
    editTextForm
      .find("button")
      .at(0)
      .simulate("click");
    expect(componentProps.handleEditText).toHaveBeenCalled();
  });

  it("button 2 should call handleDiscardText", () => {
    editTextForm
      .find("button")
      .at(1)
      .simulate("click");
    expect(componentProps.handleDiscardText).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    expect(editTextForm).toMatchSnapshot();
  });
});
