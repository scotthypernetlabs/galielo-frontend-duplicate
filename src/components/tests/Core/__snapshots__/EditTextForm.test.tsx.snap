// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTextForm Component should match snapshot 1`] = `ReactWrapper {}`;
